[DOC] Improve the hostname verification docs for NodePort listeners (#10880)
strimzi/strimzi-kafka-operator: documentation/modules/deploying/con-hostname-verification-node-ports.adoc

= Troubleshooting TLS hostname verification with node ports

[role="_abstract"]
Off-cluster access using node ports with TLS encryption enabled does not support TLS hostname verification.

> The reason is that Strimzi does not know what will be the address of the node where the broker Pod will be scheduled and cannot add it to the broker certificate.

Reviewer: ```suggestion
This is because Strimzi does not know the address of the node where the broker pod is scheduled and cannot include it in the broker certificate.
```

When configuring the client directly in Java, set the configuration option to an empty string:

[source,java]
props.put("ssl.endpoint.identification.algorithm", "");

> Alternatively, if you know the addresses of the worker nodes where the brokers will be scheduled upfront (for example because your cluster is running on bare-metal with a limited number of available worker nodes), you can use the link:{BookURLConfiguring}#property-listener-config-altnames-reference[`alternativeNames` field ^] to add additional SANs to the broker certificates manually.

Reviewer: ```suggestion
Alternatively, if you know the addresses of the worker nodes where the brokers are scheduled, you can add them as additional SANs (Subject Alternative Names) to the broker certificates manually. 
For example, this might apply if your cluster is running on a bare metal deployment with a limited number of available worker nodes. 
Use the link:{BookURLConfiguring}#property-listener-config-altnames-reference[`alternativeNames` property ^] to specify these SANS.
```
